pkg/reconciler/backup/volumesnapshot/reconciler.go

```diff
@@ -123,10 +123,20 @@ func (se *Reconciler) Execute(
 	targetPod *corev1.Pod,
 	pvcs []corev1.PersistentVolumeClaim,
 ) (*ctrl.Result, error) {
+	if cluster.Spec.Backup == nil || cluster.Spec.Backup.VolumeSnapshot == nil {
```

Comment:
we don't need to check the `Backup` here, the previous code handles this already, 
please refer to https://github.com/cloudnative-pg/cloudnative-pg/blob/main/controllers/backup_controller.go#L194

Reply:
The problem is that the unit tests won't go through that previous checking.
If you take it out and you test the code with a cluster with no backup, you get a panic.

Reply:
thanks, @jsilvela , it makes sence